app/Http/Controllers/v1/StoreUserEnrollmentController.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Http\Controllers\v1;
+
+use App\Http\Controllers\Controller;
+use App\Http\Resources\EnrollmentResource;
+use App\Http\Resources\PaginateResource;
+use App\Models\Course;
+use App\Models\Enrollment;
+use App\Services\CanvasServices;
+use Illuminate\Http\Request;
+
+class StoreUserEnrollmentController extends Controller
+{
+    /**
+     * Handle the incoming request.
+     */
+    public function __invoke(Request $request, int $providerId, string $userId)
+    {
+
+        $this->canvasService = CanvasServices::byProviderId($providerId);
+        $canvasEnrollments = $this->canvasService->getEnrollmentsByUser($userId);
+        $enrollmentCollection = collect();
+        foreach ($canvasEnrollments as $enrollment) {
+            if ($course = Course::where('provider_resource_id', $enrollment->course_id)->firstOrFail()) {
+                $request->merge([
+                    'user_id' => 1,
+                    'course_id' => $course->id,
+                    'provider_user_id' => $enrollment->user_id,
+                    'provider_course_id' => $enrollment->course_id,
+                    'provider_id' => $providerId,
+                    // 'provider_course_name' => $enrollment->sis_course_id,
+                    'enrollment_state' => $enrollment->enrollment_state,
```

Comment:
Should there be a provider_resource_id storing the enrollment id from canvas? (If there is one of course.) And could we then validate for uniqueness? Right now it looks like if I ran this action twice it would create duplicates.